packages/components/src/__tests__/utils/importDataHelpers.test.tsx

```diff
@@ -0,0 +1,158 @@
+      vc: {
+        '@context': ['https://www.w3.org/ns/credentials/v2', 'https://www.w3.org/ns/credentials/examples/v2'],
+        type: 'EnvelopedVerifiableCredential',
+        id: 'data:application/vc-ld+jwt,jwt.abc.123',
+      },
+    };
+    const credentialPath = '/vc';
+
+    (verifyVC as jest.Mock).mockImplementation(() => ({
+      verified: true,
+    }));
+    (decodeEnvelopedVC as jest.Mock).mockReturnValue({
+      type: ['VerifiableCredential'],
+      credentialSubject: {
+        id: 'did:example:123',
+        name: 'Alice',
+      },
+    });
+    const result = await processVerifiableCredentialData(data, vcContext, credentialPath);
+
+    expect(result).toEqual({
+      vc: {
+        vc: {
+          '@context': ['https://www.w3.org/ns/credentials/v2', 'https://www.w3.org/ns/credentials/examples/v2'],
+          type: 'EnvelopedVerifiableCredential',
+          id: 'data:application/vc-ld+jwt,jwt.abc.123',
+        },
+        decodedEnvelopedVC: {
+          type: ['VerifiableCredential'],
+          credentialSubject: {
+            id: 'did:example:123',
+            name: 'Alice',
+          },
+        },
+      },
+    });
+  });
+
+  it('should return input data with decodedEnvelopedVC when the VC is envelopedVC and credentialPath is not provided', async () => {
+    const data = {
+      '@context': ['https://www.w3.org/ns/credentials/v2', 'https://www.w3.org/ns/credentials/examples/v2'],
+      type: 'EnvelopedVerifiableCredential',
+      id: 'data:application/vc-ld+jwt,jwt.abc.123',
+    };
+
+    (verifyVC as jest.Mock).mockImplementation(() => ({
+      verified: true,
+    }));
+    (decodeEnvelopedVC as jest.Mock).mockReturnValue({
+      type: ['VerifiableCredential'],
+      credentialSubject: {
+        id: 'did:example:123',
+        name: 'Alice',
+      },
+    });
+    const result = await processVerifiableCredentialData(data, vcContext);
+
+    expect(result).toEqual({
+      vc: {
+        '@context': ['https://www.w3.org/ns/credentials/v2', 'https://www.w3.org/ns/credentials/examples/v2'],
+        type: 'EnvelopedVerifiableCredential',
+        id: 'data:application/vc-ld+jwt,jwt.abc.123',
+      },
+      decodedEnvelopedVC: {
+        type: ['VerifiableCredential'],
+        credentialSubject: {
+          id: 'did:example:123',
+          name: 'Alice',
+        },
+      },
+    });
+  });
+
+  it('should throw error when VC is not verified', async () => {
+    const data = {
+      vc: {
+        type: ['VerifiableCredential'],
+        credentialSubject: {
+          id: 'did:example:123',
+          name: 'Alice',
+        },
+      },
+    };
+    const credentialPath = '/vc';
+
+    (verifyVC as jest.Mock).mockImplementation(() => ({
+      verified: false,
+    }));
+    (decodeEnvelopedVC as jest.Mock).mockReturnValue(null);
+
+    expect(async () => await processVerifiableCredentialData(data, vcContext, credentialPath)).rejects.toThrow(
+      'Invalid Verifiable Credential!',
+    );
```

Comment:
I noticed that you’ve changed the error message thrown when `processVerifiableCredentialData` falls into the catch block. I think this test case should be updated to expect the error message to match the new version.